### DISCUSSION TITLE
username = admin, can this be changed?

### DISCUSSION THREAD

**Discussion by osworx** [arendst/Tasmota#17584] [Q&A]
After reading endless docs etc. all finally end with **user = admin**.
Well, fine - not really the best option for an username .. (but better than nothing)?!

So my question: is **admin** always set?
And if (or if not) can this be changed?
In the web-UI there is only a field for the password.

Need that just for making simple http requests as the docs explains them here: https://tasmota.github.io/docs/Commands/#with-web-requests

**Comment by rapi3** [comment]
EDIT: you can define another username for user admin.
Doc specify you can define in what mode interface will start and also password at compilation and also WEB_USERNAME to change default username admin.
#define WEB_SERVER             2                 // [WebServer] Web server (0 = Off, 1 = Start as User, 2 = Start as Admin)
#define WEB_PASSWORD           ""                // [WebPassword] Web server Admin mode Password for WEB_USERNAME (empty string = Disable)
  #define WEB_USERNAME         "admin"           // Web server Admin mode user name

**Comment by barbudor** [comment]
You can override the default username by redefining it in your `user_config_override.h`
Obviously that means you need to compile your own binary by any of the supported methods

So just add in your `user_config_override.h`:
```
#ifdef WEB_USERNAME         
#undef WEB_USERNAME        
#endif
#define WEB_USERNAME         "whateveryouwant"
```

**Comment by osworx** [comment]
Many thx for your answers.
But as far as I can see, they require either a new compilation or the usage of the cloud.
Both are currently no options.

But as I "discovered" right now, the console can be used to show or reset or change - is this correct?
And there for example:

> WebPassword | Show current web server password
0 = disable use of password for web UI
1 = reset password to firmware default (WEB_PASSWORD)
< value > = set web UI password (32 char limit) for user WEB_USERNAME (Default WEB_USERNAME = admin)

btw: could someone explain me in short the difference between user and admin mode?

**Reply by sfromis** [reply]
Nothing about requiring "the usage of the cloud", not sure where you got that from. But yes, you can only change the default user name with a new build. Pretty easy to do... 

User mode (unrelated to the web password) is that that web UI is strongly cut down to only include basic controls, no configuration, or ability to return to admin mode from the web UI. The console is not available. You can still do it all via MQTT and commands.

Note that in both cases, Tasmota does not profess any "hard security", and should not be opened to the Internet. Also beware that web requests are subjected to the same restrictions as the web UI.

**Comment by osworx** [comment]
@sfromis Thx for the explaination.

Mentioned the cloud because I want to call Tasmota devices only via http - which works perfectly as long as no password is set.
If a password is defined and webaccess is restricted, it does also not work if username:password is submitted .. (javascript or php cURL) - no matter which document tell me about that to use http://username:password@IP_ADDRESS/status (for example).
Shame that such does not work.

And no, I don't want to use also no MQTT or anything else.
I have homeassistent already for all everything else.

Finally it is a simple webbased display for my wife where she can see all items if they are on or off and can switch them if requried - using one of the local webservers which are running 24 hours ..

And no, nothing of all of them is "opened to the internet" - thanks.

**Reply by barbudor** [reply]
User/password on http request works if you just read the doc
It doesn't use Basic Authentication but user and password must be passed as parameter to the url

**Comment by osworx** [comment]
@barbudor Okay, my own fault - had a faulty mixed URL (based on the params I handover ..).
Now it works also with username:password (if defined) and none (if not defined).

Thanks all for your patience.
